homeassistant/components/tado/water_heater.py

```diff
@@ -311,3 +275,39 @@ def _control_heater(
             device_type=TYPE_HOT_WATER,
         )
         self._overlay_mode = self._current_tado_hvac_mode
+
+
+def _generate_entities(tado: TadoConnector) -> list[TadoWaterHeater]:
```

Comment:
Why do we move this code around? It makes the diff harder to read. Unless required here please do such moves in a separate PR.

Reply:
I moved it becuse, previously it was declaring a different return type(which was the wrong type). But now it is TadoWaterHeater and I moved methods below the definition of TadoWaterHeater

Reply:
LGTM - but see comment from Martin. 

He's not objecting to the change - just requesting it could be in a separate PR.

Reply:
Oh, I see. 

You can keep the code in its original location, still.

Options are:
 - add `from __future__ import annotations` to teh beginning of teh module, or
 - the integration is not fully typed, so could you simply remove the type hint for now
 -

Reply:
```python
# from __future__ import annotations

class TestCase1:
    pass

def test_fnc(x: "TestCase2") -> TestCase1:
    return x

class TestCase2:
    pass
```

Reply:
> Oh, I see. 
> 
> You can keep the code in its original location, still.
> 
> Options are:
>  - add `from __future__ import annotations` to the beginning of the module, or
>  - the integration is not fully typed, so could you simply remove the type hint for now, or
>  - I think you can put quotes around the class name (see below)

Don't go for full typing on the water heater. There's an incompatibly with the underlying library and HA's climate entity.
This can only be mitigated with an upgrade of the library, which is a work in progress.

Reply:
removed the typing and also moved methods to their original location